api-put-object-copy.go

```diff
@@ -49,6 +52,11 @@ func (c Client) CopyObjectWithProgress(dst DestinationInfo, src SourceInfo, prog
 		header.Set(amzLegalHoldHeader, dst.opts.LegalHold.String())
 	}
 
+	if dst.opts.Mode != RetentionMode("") && dst.opts.RetainUntilDate != timeSentinel {
```

Comment:
`time.Time` values should not be compared with `==`. Use `a.Equal(b)` or `a.IsZero()` which will be the value if it wasn't set.